content/docs/2.0/concepts/scaling-deployments.md

```diff
@@ -32,38 +40,52 @@ metadata:
       resourceMetrics:
-      - name: cpu/memory # Name of the resource to be targeted
+      - name: cpu/memory                             # Name of the resource to be targeted
         target:
           type: value/ utilization/ averagevalue
-          value: 60 # Optional
-          averageValue: 40 # Optional
-          averageUtilization: 50 # Optional
-      behavior:
+          value: 60                                  # Optional
+          averageValue: 40                           # Optional
+          averageUtilization: 50                     # Optional
+      behavior:                                      # Optional. Use to modify HPA's scaling behavior
+        scaleDown:
+          stabilizationWindowSeconds: 300
+          policies:
+          - type: Percent
+            value: 100
+            periodSeconds: 15                                     
   triggers:
-  # {list of triggers to activate the deployment}
+  # {list of triggers to activate scaling of the target resource}
 ```
 
 You can find all supported triggers [here](/scalers).
 
 ### Details
 ```yaml
   scaleTargetRef:
-    deploymentName: {deployment-name} # must be in the same namespace
-    containerName: {container-name}  #Optional. Default: deployment.spec.template.spec.containers[0]
+    apiVersion:    {api-version-of-target-resource}  # Optional. Default: apps/v1
+    kind:          {kind-of-target-resource}         # Optional. Default: Deployment
+    name:          {name-of-target-resource}         # Mandatory. Must be in the same namespace as the ScaledObject
+    containerName: {container-name}                  # Optional. Default: .spec.template.spec.containers[0]
 ```
 
-The name of the deployment this scaledObject is for. This is the deployment KEDA will scale up and setup an HPA for based on the triggers defined in `triggers:`. Make sure to include the deployment name in the label as well, otherwise the metrics provider will not be able to query the metrics for the scaled object and 1-n scale will be broken.
+The reference to the resource this ScaledObject is configured for. This is the resource KEDA will scale up/down and setup an HPA for, based on the triggers defined in `triggers:`. 
+
+To scale Kubernetes Deployments only `name` is needed to be specified, if one wants to scale a different resource such as StatefulSet or  Custom Resource (that defines `/scale` subresource), appropriate `apiVersion` (following standard Kubernetes convetion, ie. `{api}/{version}`) and `kind` need to be specfied.
 
-**Assumptions:** `deploymentName` is in the same namespace as the scaledObject
+`containerName` is a name of container in the target resource, from which KEDA should try to get environment properties holding secrets etc.
+
+**Assumptions:** Resource referenced by `name` (and `apiVersion`, `kind`) is in the same namespace as the scaledObject
```

Comment:
Should we put it like this or be explicit and say we only support scaling workloads in the same namespace as the SO?

Reply:
What do you propose by explicit?